pmm-app/src/pmm-add-instance/panel.tsx

```diff
@@ -1,48 +1,54 @@
-import React, { useState } from 'react';
-import { Button } from 'antd';
+import React, { useEffect, useMemo, useState } from 'react';
 import { createBrowserHistory } from 'history';
 import { Router, Route } from 'react-router-dom';
+import { Button } from '@grafana/ui';
+import { cx } from 'emotion';
 import AddRemoteInstance from './components/AddRemoteInstance/AddRemoteInstance';
-import DiscoveryPanel from './components/DiscoveryPanel/DiscoveryPanel';
+import Discovery from './components/Discovery/Discovery';
 import { AddInstance } from './components/AddInstance/AddInstance';
-import './panel.scss';
-import '../shared/style.less';
-import '../shared/styles.scss';
+import { getStyles } from './panel.styles';
+import { Messages } from './components/AddRemoteInstance/AddRemoteInstance.messages';
 
+const availableInstanceTypes = ['rds', 'postgresql', 'mysql', 'proxysql', 'mongodb', 'proxysql'];
 const history = createBrowserHistory();
+
 const AddInstancePanel = () => {
+  const styles = getStyles();
+
   const urlParams = new URLSearchParams(window.location.search);
   const instanceType = urlParams.get('instance_type') || '';
-  const availableInstanceTypes = ['rds', 'postgresql', 'mysql', 'proxysql', 'mongodb', 'proxysql'];
   const [selectedInstance, selectInstance] = useState({
     type: availableInstanceTypes.includes(instanceType) ? instanceType : '',
   });
 
-  const setSelectedInstance = (instance) => {
+  useEffect(() => {
     const url = new URL((window.location as unknown) as string);
 
-    url.searchParams.set('instance_type', instance.type);
-    selectInstance(instance);
+    url.searchParams.set('instance_type', selectedInstance.type);
     history.push(url.pathname + url.search);
-  };
+  }, [selectedInstance]);
+
+  const InstanceForm = useMemo(
```

Comment:
If it's used only once, I don't think it makes sense to memoize it.

Reply:
done